File: ECommerce.Web/ClientApp/app/components/shared/product.ts
export class Product {
    id: number;
    title: string;
    description: string;
    price: number;
    categoryId: number;
    categoryName: string;

    reviewList: Array<ReView> = new Array<ReView>();
}

export class ReView {
    id: string;
    reviewContent: string;
    reviewerFullName: string;
    createDateTimeOffset:Date;
}

File: ECommerce.Core/Entities/Product.cs
using System.Collections.Generic;
using CommonShares.Entities;

namespace ECommerce.Core.Entities
{
    public class Product : Entity<int>
    {
        public string Title { get; set; }
        public string Description { get; set; } 
        public decimal Price { get; set; }

        public int CategoryId { get; set; }
        public Category Category { get; set; } 

        public List<Review> ReviewList { get; set; }=new List<Review>();

    }
}
